src/engine/engine.h

```diff
@@ -17,13 +19,27 @@ struct sEngineConfiguration {
     glm::vec3 camera_center = { 0.0f, 0.75f, 0.0f };
     uint8_t msaa_count = 1;
     bool fullscreen = false;
+
+    void (*engine_post_initialize)(void) = nullptr;
```

Comment:
We should have typedefs for all those function pointers and then use that throughout the code instead of writing the verbose function pointer types everywhere.